runtime/runtime-common/src/main/kotlin/pbandk/Message.kt

```diff
@@ -28,6 +28,13 @@ interface Message<T : Message<T>> {
         }
     }
 
+    abstract class OneOf<T> {
+        abstract val value: T
+        override fun equals(other: Any?) = other is Message.OneOf<*> && value == other.value
```

Comment:
🔧 As discussed, I think this equality test is too liberal. If you have a message like:

```proto
message Bar {
    oneof foo {
        string s1 = 1;
        string s2 = 2;
     }
}
```

Then this should fail:

```kt
val barA = Bar(foo = Bar.S1("test"))
val barB = Bar(foo = Bar.S2("test"))

barA.foo == barB.foo // should return `false`, but will return `true` with the `equals` implementation in this PR
```

I think if you change `equals` to this version then it should work correctly:

```kt
abstract class OneOf<T> {
    ...
    override fun equals(other: Any?) = this::class.isInstance(other) && value == (other as OneOf<*>).value
     ...
}
```

(or you can use `other != null && this::class == other::class` if the `isInstance()` method above requires adding a dependency on `kotlin-reflect`)

Reply:
`isInstance` looks like it is available 👍